native/source/mixpanel/detail/worker.cpp

```diff
@@ -190,6 +196,50 @@ namespace mixpanel
             }
         }
 
+        int Worker::parse_www_retry_after(nanowww::Response& response)
+        {
+            auto success = false;
+            static auto failure_count = 0;
+
+            mixpanel->log(Mixpanel::LogEntry::LL_TRACE, "/track HTTP Response Headers: \n" + response.headers()->as_string());
+            mixpanel->log(Mixpanel::LogEntry::LL_TRACE, "/track HTTP Response Body: \n" + response.content());
+
+            // Check for an HTTP Retry-After header
+            auto retry_after_header = response.get_header("Retry-After");
+            auto retry_after = 0;
+            if (!retry_after_header.empty()) {
+                retry_after = atoi(retry_after_header.c_str());
+                mixpanel->log(Mixpanel::LogEntry::LL_TRACE, "Retry After Header: " + retry_after_header);
+            }
+
+            // Check for a 5XX response code
+            auto failed = (500 <= response.status() && response.status() <= 599);
+            if (failed) {
+                failure_count++;
+                mixpanel->log(Mixpanel::LogEntry::LL_ERROR, "/track HTTP Call Failed - Status Code (" + std::to_string(response.status()) + "): " + response.content());
+            } else {
+                failure_count = 0;
+                success = true;
+            }
+
+            // Calculate exponential back off
+            if (failure_count > 1) {
+                retry_after = fmax(retry_after, calculate_back_off_time(failure_count));
+                mixpanel->log(Mixpanel::LogEntry::LL_INFO, "Adding exponential back off time of " + std::to_string(retry_after) + " seconds.");
+            }
+
+            auto allowed_after_time = time(0) + retry_after;
+            mixpanel->log(Mixpanel::LogEntry::LL_TRACE, "Network requests allowed after time " + std::to_string(allowed_after_time) +
+                          ". Current time " + std::to_string(time(0)) + ". Delta: " + std::to_string(time(0) - allowed_after_time));
+            return allowed_after_time;
+        }
+
+        int Worker::calculate_back_off_time(int failure_count)
+        {
+            auto back_off_time = powf(2.0, failure_count - 1) * 60.0 + rand() % 30;
```

Comment:
On the iOS SDK we have:

    NSTimeInterval time = pow(2.0, failureCount - 1) * 60 + arc4random_uniform(30);
    return MIN(MAX(60, time), 600);

Perhaps we should standardize on something together?